app/controllers/payment-links/get-edit-amount.controller.js

```diff
@@ -21,14 +21,17 @@ module.exports = function showEditAmountPage (req, res) {
 
   const amountType = recovered.type || sessionData.price ? 'fixed' : 'variable'
   const amountInPence = recovered.amount || sessionData.price
+  const amountHint = recovered.hint || sessionData.amountHint
   const isWelsh = sessionData.language === supportedLanguage.WELSH
 
   const pageData = {
     amountType,
     amountInPence,
+    amountHint,
     isWelsh,
     errors: recovered.errors,
-    isEditing: true
+    isEditing: true,
+    displayFuturePaymentLinksContent: process.env.PAYMENT_LINKS_FUTURE_ENABLED === 'true' || sessionData.newPaymentLinkJourneyEnabled
```

Comment:
where does this `newPaymentLinkJourneyEnabled` get set or come from?

Reply:
Yeh this is a bit of a tricky one to follow through. When you're editing an existing payment link, we take the whole product and basically set `session.editPaymentLinkData` equal to the product object. So this comes from the Product we get from products

Reply:
Err I don't follow, especially the "So this comes from the Product we get from products" bits. Are there any links?

Reply:
It's done here.
https://github.com/alphagov/pay-selfservice/blob/master/app/controllers/payment-links/get-edit.controller.js#L50

It's a little bit confusing, it gets the product from products
```const product = await productsClient.product.getByProductExternalId(gatewayAccountId, productExternalId)```

Then merges it with some other data
```pageData.product = lodash.merge(product, editPaymentLinkData)```

Then sets it on the session
```lodash.set(req, 'session.editPaymentLinkData', pageData.product)```

So in the end, `session.editPaymentLinkData` contains all the fields on a GET product response. 

And the response from products is transformed using this `Product` class https://github.com/alphagov/pay-selfservice/blob/master/app/models/Product.class.js#L24

Although just spotted, we aren't loading in that `newPaymentLinkJourneyEnabled` for that class, so I'll make a small change to update that now.

Reply:
Updated

Reply:
> Although just spotted, we aren't loading in that newPaymentLinkJourneyEnabled for that class, so I'll make a small change to update that now.

I think that's it 😄